### PR TITLE
Removed ModifiedClasspathRunner and  used the ClasspathExtensions from the Spring boot test support

### PR DESCRIPTION
**Pull request by shivakrishnaah**

### CODE REVIEW

**Comment by shivakrishnaah** [comment]
Copied the same classes as in [spring-boot-testsupport](https://github.com/spring-projects/spring-boot/tree/main/spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/classpath)

**Comment by shivakrishnaah** [comment]
Fixes gh-973 as well as we migrated _ModifiedClassPathRunner_ as well

File: spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathClassLoader.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+
+import org.springframework.core.annotation.MergedAnnotation;
+import org.springframework.core.annotation.MergedAnnotations;
+import org.springframework.util.AntPathMatcher;
+import org.springframework.util.ConcurrentReferenceHashMap;
+import org.springframework.util.ObjectUtils;
+import org.springframework.util.StringUtils;
+
+/**
```
Comment (OlgaMaciaszek) [inline spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathClassLoader.java:63]: For any ported classes, please provide links to the original in javadocs.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @shivakrishnaah, thanks for submitting the PR. Have added a comment - please address it. Also, please add your full name and surname with `@author` tag to the javadocs of the classes you've modified.

File: spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathRunner.java [hunk 2]
```diff
@@ -1,362 +0,0 @@
...
- * A custom {@link BlockJUnit4ClassRunner} that runs tests using a modified class path.
- * Entries are excluded from the class path using
- * {@link ClassPathExclusions @ClassPathExclusions} and overridden using
- * {@link ClassPathOverrides @ClassPathOverrides} on the test class. A class loader is
- * created with the customized class path and is used both to load the test class and as
- * the thread context class loader while the test is being run.
- *
- * @author Andy Wilkinson
- */
-public class ModifiedClassPathRunner extends BlockJUnit4ClassRunner {
```
Comment (OlgaMaciaszek) [inline spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathRunner.java:71]: I've noticed this was a `public` class, so we cannot just remove it in a non-major release. Please mark it as `@Deprecated` and add a `@deprecated` tag with `@link` to the new implementation.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks, @shivakrishnaah. Have noticed one more thing that needs to be addressed. Please take a look at the comment.

File: spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathRunner.java (hunk 2 above)
Comment (shivakrishnaah) [reply spring-cloud-test-support/src/main/java/org/springframework/cloud/test/ModifiedClassPathRunner.java:71]: 👍🏼

**Review by OlgaMaciaszek — APPROVED** [review]
Thanks, @shivakrishnaah. LGTM.

**Pull request merged**